.travis.yml

```diff
@@ -12,23 +14,42 @@ before_install:
   - export PATH=$PWD/apache-maven-3.3.9/bin:${PATH}
   - hash -r
 
-cache:
-  directories:
-    - $HOME/.m2
-
-jdk:
-  - openjdk7
-  - oraclejdk7
-  - oraclejdk8
-
-
-# Stage 0: Test -  Builds against all jdks
-script: mvn -q install
-
-
 jobs:
   include:
-    # Stage 1: Deploy - Builds and deploys all artifacts with java 8.
+    # Stage Test - Job 1-3:  Builds against all Jdks.
+    - stage: test
+      jdk: openjdk7
+      script: mvn -q install
+    - jdk: oraclejdk7
+      script: mvn -q install
+    - jdk: oraclejdk8
+      script: mvn -q install
+    # Stage Test - Job 4: Build Android Examples.
+    - jdk: oraclejdk8
+      env: ANDROID_EXAMPLES=true
+      language: android
+      android:
+        components:
+          - tools
+          - platform-tools
+          - tools # Note that the tools section appears twice on purpose as it’s required to get the newest Android SDK tools.
+          - build-tools-26.0.0
+          - android-21
+          - sys-img-armeabi-v7a-android-21
+      before_install:
+        # Keep android version in sync with  platform.sdk in android-examples.pom.
+        - echo no | android create avd --force -n test -t android-21 --abi armeabi-v7a
+        # Create and start the emulator. This takes some time.
+        - emulator -avd test -no-skin -no-audio -no-window &
+      install:
+        # Build the dependencies for the android examples while the emulator starts.
+        - mvn -q install --also-make --projects android,picocontainer -DskipTests=true -Dmaven.javadoc.skip=true
+      before_script:
+        - android-wait-for-emulator
+      script:
+        - mvn -q install --activate-profiles android-examples --projects examples/android --also-make-dependents -Dandroid.device=test
```

Comment:
One drawback of using `mvn -q` when actually running the android tests, is that if they are not finished in 10 minutes [Travis will stop the build](https://travis-ci.org/cucumber/cucumber-jvm/jobs/254050868).